miden-lib/src/transaction/procedures.rs

```diff
@@ -1,9 +1,22 @@
 use alloc::vec::Vec;
 
-use miden_objects::{digest, Digest, Felt, Hasher};
+use miden_objects::{utils::Deserializable, Digest, Felt, Hasher};
+use once_cell::sync::OnceCell;
 
 use super::TransactionKernel;
 
+// CONSTANTS
+// ================================================================================================
+
+/// Number of currently used kernel versions.
+const NUM_VERSIONS: usize = 1;
+
+/// Include file with kernel 0 procedure hashes generated in build.rs
+const PROCEDURES_RAW: [&[u8]; NUM_VERSIONS] = [include_bytes!("../../kernel_procs_v0.bin")];
+
+/// Array of all available kernels.
+pub static PROCEDURES: [OnceCell<Vec<Felt>>; NUM_VERSIONS] = [OnceCell::new()];
```

Comment:
Will it be better to store these constants as a part of the `TransactionKernel` struct?

Reply:
Unfortunately, we can't put statics into `impl` blocks. But we're going to move back to generation of rust sources, so we will be able to use constants, as in previous implementation.